src/panels/lovelace/mixins/element-click-mixin.js

```diff
@@ -12,8 +13,84 @@ import computeStateName from "../../../common/entity/compute_state_name";
 export default dedupingMixin(
   (superClass) =>
     class extends NavigateMixin(EventsMixin(superClass)) {
-      handleClick(hass, config) {
-        const tapAction = config.tap_action || "more-info";
+      registerMouse(config) {
+        let ripple = null;
+        const rippleWrapper = document.createElement("div");
+        this.shadowRoot.appendChild(rippleWrapper);
+        rippleWrapper.style.position = "absolute";
+        rippleWrapper.style.width = "100px";
+        rippleWrapper.style.height = "100px";
+        rippleWrapper.style.top = "50%";
+        rippleWrapper.style.left = "50%";
+        rippleWrapper.style.transform = "translate(-50%, -50%)";
+        rippleWrapper.style.pointerEvents = "none";
+
+        const loadRipple = () => {
+          if (ripple) return;
+          ripple = document.createElement("mwc-ripple");
+          rippleWrapper.appendChild(ripple);
+          ripple.unbounded = true;
+          ripple.primary = true;
+        };
+        const startAnimation = () => {
+          ripple.style.visibility = "visible";
+          ripple.disabled = false;
+          ripple.active = true;
+        };
+        const stopAnimation = () => {
+          if (ripple) {
+            ripple.active = false;
+            ripple.disabled = true;
+            ripple.style.visibility = "hidden";
+          }
+        };
+
+        var isTouch =
+          "ontouchstart" in window ||
+          navigator.MaxTouchPoints > 0 ||
+          navigator.msMaxTouchPoints > 0;
+        var mouseDown = isTouch ? "touchstart" : "mousedown";
+        var mouseOut = isTouch ? "touchcancel" : "mouseout";
+        var click = isTouch ? "touchend" : "click";
+
+        var timer = null;
+        var held = false;
+        var holdTime = config.hold_time || 500;
+
+        this.addEventListener(mouseDown, () => {
+          held = false;
+          loadRipple();
```

Comment:
The ripple needs to be lazy-loaded on mouse-down rather than when the animation happens.
The ripple can't be set active until it has finished rendering, and for some reason waiting for `ripple.updateComplete` doesn't work on iphone...